File: Assets/ToLua/Lua/user/ui/shop/model/data/shop_limit_item_data.lua
local t = {}
t.data = {}
local item = {}
item.__index = item

function item.New(table)
	local o = {}
	setmetatable(o,item)
	o.id = tonumber(table.id)
	o.sheet_num = tonumber(table.sheet_num)
	o.buy_type = tonumber(table.buy_type)
	o.type = tonumber(table.type)
	o.resource = table.resource
	o.name = table.name
	o.pic = table.pic
	o.des = table.des
	o.cost = table.cost
	o.pre_item = tonumber(table.pre_item)
	o.item_num = tonumber(table.item_num)
  o.discount = tonumber(table.discount)
	return o
end

function t.GetDataById(id)
	return t.data[id]
end

function t.GetActionItemDataList ()
  local result = {}
  for k, v in pairs(t.data) do
    -- [[ v.sheet_num == 14 表示行动力商品 ]] --
    if v.sheet_num == 14 then
      result[k] = v
    end
  end
  return result
end

function t.GetGoldItemDataList ()
  local result = {}
  for k, v in pairs(t.data) do
    -- [[ v.sheet_num == 14 表示金币商品 ]] --
    if v.sheet_num == 15 then
      result[k] = v
    end
  end
  return result
end

local function Start()
	local origin = dofile('shop_limit_item')
	origin.ForEach(function(id,table)
		local newItem = item.New(table)
		t.data[id] = newItem
	end)
end

Start()
return t

File: Assets/Code/Logic/Shop/Model/ShopLimitItemData.cs
using System.Collections.Generic;
using Common.Util;
using Logic.Enums;
using Logic.Game.Model;

namespace Logic.Shop.Model
{
	public class ShopLimitItemData
	{
		private static Dictionary<int, ShopLimitItemData> _shopLimitItemDataDictionary;
		public static Dictionary<int, ShopLimitItemData> GetShopLimitItemDataDictionary ()
		{
			if (_shopLimitItemDataDictionary == null)
			{
				_shopLimitItemDataDictionary = CSVUtil.Parse<int, ShopLimitItemData>("config/csv/shop_limit_item", "id");
			}
			return _shopLimitItemDataDictionary;
		}

		public static List<ShopLimitItemData> GetActionItemDatalist ()
		{
			List<ShopLimitItemData> actionItemDataList = new List<ShopLimitItemData>();
			List<ShopLimitItemData> shopLimitItemDataList = GetShopLimitItemDataDictionary().GetValues();
			int shopLimitItemDataCount = shopLimitItemDataList.Count;
			for (int i = 0; i < shopLimitItemDataCount; i++)
			{
				if (shopLimitItemDataList[i].shopID == 14)
				{
					actionItemDataList.Add(shopLimitItemDataList[i]);
				}
			}
			return actionItemDataList;
		}

		public static List<ShopLimitItemData> GetGoldItemDataList ()
		{
			List<ShopLimitItemData> goldItemDataList = new List<ShopLimitItemData>();
			List<ShopLimitItemData> shopLimitItemDataList = GetShopLimitItemDataDictionary().GetValues();
			int shopLimitItemDataCount = shopLimitItemDataList.Count;
			for (int i = 0; i < shopLimitItemDataCount; i++)
			{
				if (shopLimitItemDataList[i].shopID == 15)
				{
					goldItemDataList.Add(shopLimitItemDataList[i]);
				}
			}
			return goldItemDataList;
		}

		[CSVElement("id")]
		public int id;
		
		[CSVElement("sheet_num")]
		public int shopID;
		
		[CSVElement("buy_type")]
		public int buyType;

		public BaseResType baseResType;
		[CSVElement("type")]
		public int type
		{
			set
			{
				baseResType = (BaseResType)value;
			}
		}
		public GameResData resourseGameResData;
		[CSVElement("resource")]
		public string resource
		{
			set
			{
				resourseGameResData = new GameResData(value);
			}
		}

		[CSVElement("name")]
		public string name;

		[CSVElement("pic")]
		public string pic;

		[CSVElement("des")]
		public string description;

		public GameResData costGameResData;
		[CSVElement("cost")]
		public string cost
		{
			set
			{
				costGameResData = new GameResData(value);
			}
		}

		[CSVElement("pre_item")]
		public int preItem;

		[CSVElement("item_num")]
		public int itemNum;
	}	
}
